libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2750,8 +2750,16 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 			);
 
 		if (!funType->bound())
-			if (auto contractType = dynamic_cast<ContractType const*>(exprType))
-				requiredLookup = contractType->isSuper() ? VirtualLookup::Super : VirtualLookup::Virtual;
+		{
+			if (auto typeType = dynamic_cast<TypeType const*>(exprType))
```

Comment:
```suggestion
			if (auto const* typeType = dynamic_cast<TypeType const*>(exprType))
```

Reply:
The deduced type is the same in both cases. In fact the conversions `T const*` into `T*` and `T const*` into `T` are type errors, so the deduced type should be quite clear here. I would prefer not writing `const *` twice, once inside `dynamic_cast` and the other next to auto.

Reply:
Although in the codebase, both forms are used,  both `auto x = dynamic_cast<T const*>(...)` and `auto const* = dynamic_cast<T const*>(...)`

Reply:
Yea, I usually use `auto const*` to help/force some less clever compilers, but fine with either